src/main/java/gregtech/integration/jei/recipe/GTRecipeWrapper.java

```diff
@@ -76,9 +79,25 @@ public void getIngredients(@Nonnull IIngredients ingredients) {
 
         // Outputs
         if (!recipe.getOutputs().isEmpty() || !recipe.getChancedOutputs().isEmpty()) {
-            List<ItemStack> recipeOutputs = recipe.getOutputs()
-                    .stream().map(ItemStack::copy).collect(Collectors.toList());
-            currentItemSlot += recipeOutputs.size();
+            List<ItemStack> recipeOutputs;
+
+            // Scanner Output replacing
+            if (recipe.getOutputs().get(0).hasTagCompound() && recipe.getOutputs().get(0).isItemEqual(MetaItems.TOOL_DATA_STICK.getStackForm(1))) {
+                ItemStack dataStick = recipe.getOutputs().get(0);
+                NBTTagCompound researchItemNBT = dataStick.getSubCompound("asslineOutput");
+                try {
+                    ItemStack researchItem = new ItemStack(researchItemNBT);
+                    recipeOutputs = Collections.singletonList(researchItem);
+                } catch (NullPointerException e) {
+                    recipeOutputs = recipe.getOutputs()
+                            .stream().map(ItemStack::copy).collect(Collectors.toList());
+                    currentItemSlot += recipeOutputs.size();
+                }
```

Comment:
Is there any need for this try/catch at all? Creating a new ItemStack from an NBT compound should not return a null item at all. If it does, you should just check if `researchItem` is null before trying to add it to the recipe outputs, instead of using a try/catch

Reply:
testing again